src/string_bytes.cc

```diff
@@ -49,6 +49,10 @@ class ExternString: public ResourceType {
                                      size_t length) {
       EscapableHandleScope scope(isolate);
 
+      if (length > String::kMaxLength) {
+        return Local<String>();
+      }
```

Comment:
I don't think this is quite the proper fix.  It'd be better to try and create the string but release the memory when V8 returns an empty handle.